### PR TITLE
UI-225 updates to package.json to update eslint to 4.11.0 

### PR DESCRIPTION
**Pull request by joshsylvester**
* update eslint to 4.11.0 to add lines-between-class-members rule.
* synced jest & babel-core to match ui-components-predix.

### CODE REVIEW

File: packages/react-scripts/package.json [hunk 1]
```diff
@@ -38,29 +38,31 @@
...
     "node-sass": "4.5.3",
     "object-assign": "4.1.1",
     "postcss-flexbugs-fixes": "3.2.0",
     "postcss-loader": "2.0.6",
     "prettier": "^1.7.4",
     "prettier-eslint-cli": "4.1.1",
     "promise": "8.0.1",
+    "react": "16.0.0",
     "react-dev-utils": "^4.0.0",
+    "react-dom": "^16.0.0",
```
Comment (shinxi) [inline packages/react-scripts/package.json]: The react-dom is with [caret ranges](https://docs.npmjs.com/misc/semver#caret-ranges-123-025-004) while react is with exact version, not sure if it is a typo or on purpose?

Comment (joshsylvester) [reply packages/react-scripts/package.json]: mistake - i'll update them both to be the same.

Comment (joshsylvester) [reply packages/react-scripts/package.json]: updated. @shinxi please review again, thanks!

**Comment by shinxi** [comment]
The change looks good to me.

**Pull request merged**